pandas/io/pytables.py

```diff
@@ -1133,6 +1133,10 @@ def _create_storer(self, group, format=None, value=None, append=False,
                        **kwargs):
         """ return a suitable class to operate """
 
+        if isinstance(value, Series) and any(np.iscomplex(value)) and format=='table':
```

Comment:
The only reason to know it is a Series is to provide a meaningful error message - the error is the same place as the other new error message, when trying to index a complex column.